### PR TITLE
Optimizer both versions

### PR DESCRIPTION
**Pull request by perezp**
This PR creates the Seaclouds optimizer as a jar (optimizer-core) for importing and java method invocation, and also as a service with war packaging (optimizer-service).
Optimizer functionality has not changed at all in this period. 
This PR includes tests of the service running on Jetty to crate the war.
All files have changed, mainly due to the refactoring.

This PR is in conflict with the previous one of the optimizer. 
@andreaturli @adriannieto @rosogon @szenzaro @MicheleGuerriero @kiuby88 and everybody who may be interested in today's discussion during the integration telco. If any of you are happy to review the changes in this PR I discard the previous PR.

In the next PR I'll fill the part of the optimizer in the file .bom for brooklyn for its deployment with the platform.

### CODE REVIEW

File: planner/optimizer/optimizer-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>eu.seaclouds-project</groupId>
+    <artifactId>optimizer</artifactId>
+    <version>0.8.0-SNAPSHOT</version>
+  </parent>
+  <groupId>eu.seaclouds-project</groupId>
+  <artifactId>optimizer-core</artifactId>
+  <version>0.8.0-SNAPSHOT</version>
```
Comment (andreaturli) [inline planner/optimizer/optimizer-core/pom.xml]: remove this version

File: planner/optimizer/optimizer-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  <name>optimizer-core</name>
+  <url>http://maven.apache.org</url>
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+  <dependencies>
+
+
```
Comment (andreaturli) [inline planner/optimizer/optimizer-core/pom.xml]: weird indentation

File: planner/optimizer/pom.xml [hunk 3]
```diff
@@ -18,27 +18,27 @@
...
-		</dependency>
 		<dependency>
 			<groupId>org.slf4j</groupId>
 			<artifactId>slf4j-api</artifactId>
 		</dependency>
 		<dependency>
-			<groupId>org.testng</groupId>
-			<artifactId>testng</artifactId>
-			<scope>test</scope>
+			<groupId>org.slf4j</groupId>
```
Comment (andreaturli) [inline planner/optimizer/pom.xml]: remove this dependency: see https://github.com/SeaCloudsEU/SeaCloudsPlatform/commit/872e6c5d3ff6f2eddc9e06dd4a6c6547390f5918

File: planner/pom.xml [hunk 4]
```diff
@@ -44,7 +44,7 @@ limitations under the License.
 
     </dependencies>
 
-    <repositories>
+     <repositories>
```
Comment (andreaturli) [inline planner/pom.xml]: is it still needed?

**Comment by rosogon** [comment]
@perezp : It looks like optimizer-core and src have the same files. I guess you copied the files from src to optimizer-core and forgot to remove in src. If this was your intention, the good state is that git realizes you are moving the files, so they don't appear as new content. The PR would be much easier to review, because it will only contain the service code.

**Comment by perezp** [comment]
@rosogon Thank you for the comment! 

I did my best with eclipse using only the "move" and "rename" options... but something could have failed or I made also some raw copy. Actually I didn't know it was duplicated. One of the parts is useless because it is code that is in a project with "pom" packaging.

File: planner/optimizer/optimizer-core/pom.xml (hunk 1 above)
Comment (perezp) [inline planner/optimizer/optimizer-core/pom.xml]: Done, thanks!
Change not pushed yet.

File: planner/optimizer/optimizer-core/pom.xml (hunk 2 above)
Comment (perezp) [inline planner/optimizer/optimizer-core/pom.xml]: Done. Thanks!

File: planner/optimizer/pom.xml (hunk 3 above)
Comment (perezp) [inline planner/optimizer/pom.xml]: Done, Thanks!

File: planner/pom.xml (hunk 4 above)
Comment (perezp) [inline planner/pom.xml]: It was not my intention to change the pom.xml of planner. It looks like a white space was accidentally included.

**Comment by perezp** [comment]
@andreaturli @rosogon 
I've done the changes you have proposed. Thank you very much. All of them were necessary. 
I've already pushed the new version including your suggestions. Let me know.

**Comment by andreaturli** [comment]
@perezp I think all the yaml file like `src/test/java/eu/seaclouds/platform/planner/optimizerTest/resources/MMoutputCleanYAMLV-15-09.yml` belong to `src/test/resources`: they are not java tests so better to use the usual maven layout. 

wdyt?

**Comment by perezp** [comment]
@andreaturli Fine, I change it. I tried to include everything I did under some "..../planner/optimizer/...." path to avoid mixing things. Maybe it was not the right decision. I replace it with the usual maven layout as you suggest.

**Comment by andreaturli** [comment]
thanks @perezp

**Pull request merged**